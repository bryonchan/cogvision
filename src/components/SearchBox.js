import React, { PropTypes } from 'react';
import {FormGroup, FormControl, HelpBlock} from 'react-bootstrap';

        // <input {...props} />
const SearchBox = ({...props}) => {
    return (
    	<FormGroup
          controlId="formBasicText"
          validationState={props.errorMessage ? "error" : null}
        >
	        <FormControl
	            type="text"
	            placeholder={props.placeholder}
	            onChange={props.onChange}
	          />
	          {props.errorMessage && 
		          <HelpBlock>{props.errorMessage}</HelpBlock>
		      }
		      {props.totalNumberOfResults && 
		          <HelpBlock>Total found: {props.totalNumberOfResults}</HelpBlock>
		      }
        </FormGroup>
    );
};

SearchBox.displayName = 'SearchBox';

SearchBox.propTypes = {
    className: PropTypes.string,
    placeholder: PropTypes.string,
};

export default SearchBox;
