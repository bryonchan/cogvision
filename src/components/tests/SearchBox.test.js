import React from 'react';
// import Link from '../Link.react';
import SearchBox from '../SearchBox';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <SearchBox placeholder="Search..."></SearchBox>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});