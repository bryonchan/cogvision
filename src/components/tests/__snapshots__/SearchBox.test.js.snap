exports[`test renders correctly 1`] = `
<div
  className="form-group">
  <input
    className="form-control"
    id="formBasicText"
    onChange={undefined}
    placeholder="Enter text"
    type="text" />
</div>
`;
